discord_bot.py

```diff
@@ -218,7 +218,7 @@ async def background_task():
                 stdout=asyncio.subprocess.PIPE,
                 stderr=asyncio.subprocess.PIPE)
 
-            stdout, stderr = await proc.communicate()
+            stdout, stderr = await asyncio.wait_for(proc.communicate(), 30)
             output = stdout.decode()
             if not output:
                 continue
```

Comment:
Based on the first example in the documentation, we should be checking `stdout` instead of the result of `decode()`.
https://docs.python.org/3/library/asyncio-subprocess.html

```suggestion
            if not stdout:
                continue
            output = stdout.decode()
```

Reply:
I went with a try catch around the `wait_for` since that is what i see in examples.

Reply:
Yeah, we do need the try/catch. I convinced myself that the way we check `stdout` is probably okay. This comment can be resolved.